### DISCUSSION TITLE
Is it possible to disable rocket logs completely?

### DISCUSSION THREAD

**Discussion by Leonezz** [rwf2/Rocket#2897] [Questions] [answered]
I am trying to use rocket in my project as a http server component, and I would like to log only events that happened in my application level, instead of the framework. I have set `log_level` to `off` to disable the normal logs out from rocket, but if I start my server with `LOG_LEVEL=trace` envrionment, rocket still output its internal logs.

**Comment by the10thWiz** [comment]
Assuming you're using a logger other than the rocket default, you will need to explicitly disable the logs rocket generates. This can be done with `LOG_LEVEL=trace,rocket=off`, assuming you are using `env_logger`.

**Reply by Leonezz** [reply]
> Assuming you're using a logger other than the rocket default, you will need to explicitly disable the logs rocket generates. This can be done with `LOG_LEVEL=trace,rocket=off`, assuming you are using `env_logger`.

Thank you, it's great to know! I am using tracing as logger and I wonder is there a pragmatic way to disable it, so that I don't need to explicitly set rocket related envs (as well as other crate related) for every run.

**Reply by the10thWiz** [reply, marked as answer]
You'll need to look into the logger you're using. `env_logger` supports configuring these filters in the code itself as well (when you create the logger), and I believe the tracing crates do as well.
